sink.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"bytes"
+	"container/ring"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/xmidt-org/retry"
+	"github.com/xmidt-org/retry/retryhttp"
+	"github.com/xmidt-org/wrp-go/v3"
+	"github.com/xmidt-org/wrp-go/v3/wrphttp"
+	"go.uber.org/zap"
+)
+
+type SinkI interface {
+	Update(Listener) error
+	Send(*ring.Ring, string, string, *wrp.Message) error
+}
+
+type SinkWebhookV1 struct {
```

Comment:
```suggestion
type WebhookV1 struct {
```
later on if you move this file to it's own package, e.g. `sink` package, this name might stutter. e.g. `sink.SinkWebhookV1` vs `sink.WebhookV1`